### ISSUE TITLE
Make Account Currency Configurable

### ISSUE THREAD

**Issue by jaredbroad** [QuantConnect/Lean#110]
In CashBook class there's a constant AccountCurrency, this serves as the base currency for the portfolio. To make this a configurable property:
1. IBrokerage .GetCashBalances() should return an "account currency flag"
2. ISetupHandler would use this to set the account currency for the cashbook.

**Comment by Worthy7** [comment]
Can you write this up into the documentation please

**Comment by jaredbroad** [comment]
Notes from @StefanoRaggi 

```
- Move CashBook.AccountCurrency to QCAlgorithm.AccountCurrency (instance property, not static):
 - can only be set in Initialize, locked after

- Add QCAlgorithm.SetAccountCurrency:
 - should be OK to be called in Initialize, brokerage.GetCashBalance/GetOpenOrders/GetAccountHoldings are called after Initialize
 - some brokerages (e.g. GDAX) force an account currency (EUR for European users)
 - other brokerages (Oanda) allow user to choose an account currency at account creation time (USD, EUR, ...)
 - other exchanges (Binance) do not have fiat currencies (so user should be allowed to select BTC, ETH, ...)

- SecurityPortfolioModel always subtracts fees assuming in account currency
 - should be using quote currency (or the brokerage-specific currency for fees, if different from quote currency)
 - may need to refactor IFeeModel.GetOrderFee to return an object (value+currency) instead of a decimal (this might be a separate PR to be done first)

- Depending on the account currency chosen, this issue might need to be resolved first (not required for GDAX):
https://github.com/QuantConnect/Lean/issues/2060```

**Comment by voidale** [comment]
Any update on this? :)

**Comment by voidale** [comment]
@Martin-Molinero Awesome, good luck finishing that one, definitely will be using that when it's available.

**Comment by Doggie52** [comment]
I'm assuming this would allow us to choose which currency to base our live deployment on as well? If so then this would be great news.

Correct me if I'm wrong, but currently the only "problem" with a live deployment is that the runtime statistics are reported in USD (and hence subject to <baseccy>USD fluctuations)? Otherwise it's not like orders are treated any differently if the funds are in <baseccy>, right, as that's all managed on the broker side?

**Comment by StefanoRaggi** [comment]
Closed by #4806
